Source/Parallelization/GuardCellManager.cpp

```diff
@@ -139,24 +139,30 @@ guardCellManager::Init (
     if (electromagnetic_solver_id != ElectromagneticSolverAlgo::None &&
         electromagnetic_solver_id != ElectromagneticSolverAlgo::HybridPIC)
     {
-        for (int i = 0; i < AMREX_SPACEDIM; i++)
-        {
-            amrex::Real dt_Rho = dt;
-            amrex::Real dt_J = 0.5_rt*dt;
-            if (do_multi_J) {
-                // With multi_J + time averaging, particles can move during 2*dt per PIC cycle.
-                if (fft_do_time_averaging){
-                    dt_Rho = 2._rt*dt;
-                    dt_J = 2._rt*dt;
-                }
-                // With multi_J but without time averaging, particles can move during dt per PIC
-                // cycle for the current deposition as well.
-                else {
-                    dt_J = dt;
+        const ParmParse pp_warpx("warpx");
+        bool const nguards_specified = utils::parser::queryWithParser(pp_warpx, "nguards", ng_alloc_Rho);
+        if (nguards_specified) {
+            ng_alloc_J = ng_alloc_Rho;
```

Comment:
If a users sets this parameter *smaller* than the auto-calculcated value, should we throw a warning?
I see users having this parameter in their input and silently corrupting their memory.

Can you slightly restructure for this?